sdk/go/common/resource/plugin/provider.go

```diff
@@ -101,6 +101,10 @@ type Provider interface {
 	// non-blocking; it is up to the host to decide how long to wait after SignalCancellation is
 	// called before (e.g.) hard-closing any gRPC connection.
 	SignalCancellation() error
+
+	// Attach triggers an attach for a currently running provider to the engine
+	// TODO It would be nice if this was a HostClient rather than the string address but due to dependency ordering we don't have access to declare that here.
+	Attach(address string) error
```

Comment:
Can this be an implementation detail of the gRPC implementation of plugins rather than part of the `Plugin` type?

Reply:
Maybe? I thought about maybe having a GrpcProvider that was just Provider + Attach? Is there a more "go" way to do it?

Reply:
Yeah, I'm wondering whether or not we need to expose `Attach` at all. I think we could just invoke the gRPC method directly here: https://github.com/pulumi/pulumi/pull/8979/files#diff-4372cd77587542c3e09139fb236d0bce7d0ff31c3c81864950974b1e33e5104eR146